flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpReporter.java

```diff
@@ -126,11 +126,14 @@ public void report() {
 				g.getMetricValue();
 				request.addGauge(g);
 			} catch (Exception e) {
-				// Remove that Gauge if it's not of Number type
-				gauges.remove(entry.getKey());
+				LOGGER.warn("the Gauge {} is not of Number type", g.getMetric());
```

Comment:
There are a few things I'd like to change here. First I'd like us to be able to differentiate between broken gauge implementations (throwing exceptions) and non-Number gauges (I guess resulting in a ClassCastException).

Then, log on INFO either:
`The Gauge {} will not be reported because only number types are supported by this reporter.`
`The Gauge {} will not be reported because it threw an exception.`

In case of a random exception, also log the exception on TRACE.

Reply:
@jinglining you haven't fully addressed this comment.